staging/src/k8s.io/apimachinery/pkg/watch/watch.go

```diff
@@ -195,11 +195,16 @@ func (a *AggregatedWatcher) Stop() {
 	if !a.stopped {
 		a.stopped = true
 		a.allowWatcherReset = false
+
+		a.mapLock.RLock()
```

Comment:
not sure why to write to the stop channel needs unlock the watcher map, some comment can help here.

Reply:
Updated due to CR feedback.